src/border.jl

```diff
@@ -997,6 +997,25 @@ expand(inds::Indices, kernel) = expand(inds, calculate_padding(kernel))
 expand(inds::Indices, pad::Indices) = firsttype(map_copytail(expand, inds, pad))
 expand(ind::AbstractUnitRange, pad::AbstractUnitRange) = typeof(ind)(first(ind)+first(pad):last(ind)+last(pad))
 expand(ind::Base.OneTo, pad::AbstractUnitRange) = expand(UnitRange(ind), pad)
+expand(ind::SOneTo{T₁}, pad::AbstractUnitRange) where {T₁} = represent_unit_range_with_SOneTo(ind, pad)
+expand(ind::OffsetArrays.IdOffsetRange{T₁, SOneTo{T₂}}, pad::AbstractUnitRange)  where {T₁, T₂} = determine_offset_with_SOneTo(ind, pad)
+
+function represent_unit_range_with_SOneTo(ind::SOneTo{T₁}, pad::AbstractUnitRange) where {T₁}
+    lo = first(ind)+first(pad)
+    hi = last(ind)+last(pad)
+    interval = 1:length(lo:hi)
+    T = typeof(T₁)
+    I = typeof(SOneTo(length(interval)))
```

Comment:
This is unfortunately a type-unstable patch: `SOneTo{n}` where `n = length(interval)` is runtime information.

IIUC, unless the image itself is also a SMatrix, we can't get a static `pad`, and thus we can't make it type stable.

The solution to support static array inputs might be converting it back to normal array and unit ranges.